paperqa/agents/tools.py

```diff
@@ -190,19 +190,28 @@ async def gather_evidence(self, question: str, state: EnvironmentState) -> str:
         if not state.docs.docs:
             raise EmptyDocsError("Not gathering evidence due to having no papers.")
 
+        if f"{self.TOOL_FN_NAME}_initialized" in self.settings.callbacks:
+            callback_list = self.settings.callbacks[f"{self.TOOL_FN_NAME}_initialized"]
+            for callback in callback_list:
+                await callback(state)
```

Comment:
```suggestion
            await asyncio.gather(*(c(state) for c in self.settings.callbacks[f"{self.TOOL_FN_NAME}_initialized"])
```

Let's parallelize the callback calling

Reply:
oh neat! Similar to `Promise.all`
thx!!

Reply:
Yeah it's a nice one. Cheers!